src/OpenApi/src/Services/OpenApiDocumentProvider.cs

```diff
@@ -0,0 +1,46 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.OpenApi.Writers;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.AspNetCore.OpenApi;
+using Microsoft.Extensions.Options;
+using System.Linq;
+using Microsoft.OpenApi.Extensions;
+
+namespace Microsoft.Extensions.ApiDescriptions;
+
+internal class OpenApiDocumentProvider(IServiceProvider serviceProvider) : IDocumentProvider
```

Comment:
I assume most of these classes will end up `sealed` once they're fleshed out?

Reply:
They can be sealed now. The implementation in this particular class is complete and will not likely change for the remainder of this work.